conan/tools/env/envtoolchain.py

```diff
@@ -8,13 +8,23 @@ def __init__(self, conanfile):
 
     def env(self):
         # Visit all dependencies
-        env = Environment()
-        for node in self._conanfile.deps:
-            env = env.compose(node.env)
-        return env
+        deps_env = Environment()
+        for dep in self._conanfile.dependencies.all:
+            dep_env = dep.new_env
+            if dep_env is not None:
+                deps_env = deps_env.compose(dep_env)
+
+        profile_env = Environment()
+        for k, v in self._conanfile._conan_env_values.data.items():
+            if k is None:
```

Comment:
what is these `k` being `None`? I don't remember what it means.

Reply:
That is the current ``EnvValues`` model: {pattern: {var-name: var-value}}. The pattern is the package-name pattern, being ``None``= Affect all packages.